webpack/assets/javascripts/react_app/components/HostDetails/index.js

```diff
@@ -132,25 +114,34 @@ const HostDetails = ({ match, location: { hash } }) => {
           </Text>
           <br />
         </div>
-        <Tabs
-          style={{
-            width: window.innerWidth - (isNavCollapsed ? 95 : 220),
-          }}
-          activeKey={activeTab}
-          onSelect={handleTabClick}
-        >
-          {tabs &&
-            tabs.map(tab => (
-              <Tab eventKey={tab} title={tab}>
-                <Slot
-                  response={response}
-                  status={status}
-                  id="host-details-page-tabs"
-                  fillID={tab}
-                />
-              </Tab>
-            ))}
-        </Tabs>
+        <HashRouter>
+          <>
+            <Tabs
+              style={{
+                width: window.innerWidth - (isNavCollapsed ? 95 : 220),
+              }}
+              activeKey={hash.slice(2)}
+            >
+              {tabs &&
+                tabs.map(tab => (
+                  <Tab eventKey={tab} title={tab} href={`#${tab}`} />
+                ))}
+            </Tabs>
+            <Switch>
+              {tabs?.map(tab => (
+                <Route path={`/${tab}`}>
+                  <Slot
```

Comment:
looks great, should it be: `<Route path={`/${tab}`} component={SpecificSlot}/>` ?

Reply:
using `component` will create the element on every mount, so when we have props to add we should use render instead.

For more info about the differences between the two - https://migcoder.medium.com/difference-between-render-and-component-prop-on-react-router-v4-368ca7fedbec

Reply:
Is this necessary:

Reply:
Thanks @Andrewgdewar, a key is needed here